### DISCUSSION TITLE
how to add additional TextStyle? For example when we have 20 TextStyle in figma

### DISCUSSION THREAD

**Discussion by DigitalKoi** [xmartlabs/gong#254] [Q&A] [answered]
https://github.com/xmartlabs/gong/blob/b0b617e56403c1f499704111acad89093aa3c9d6/app/src/main/java/com/xmartlabs/gong/ui/theme/AppTypographies.kt#L62

**Comment by mirland** [comment, marked as answer]
Hi! you have 2 options, it will depend if that will be used multiple times or only once.
If you have multiple TS with minor adjustments, and you will use it just once, you can change it when you are creating your composable, something like:
```kotlin
AppTheme.typography.subtitle1.copy(color = AppTheme.colors.linkTextColor)
```

If you are in the other case, if you will use it multiple times, I'd recommend you to include it in your app theme.
To do that you have to add the new text style to the `AppTypography` class. Gong has already included two custom text styles. If you want to include a new one, you have to declare [it in this class](https://github.com/xmartlabs/gong/blob/ecbd4871f87338028dd4289eebe55e2d8728b564/app/src/main/java/com/xmartlabs/gong/ui/theme/AppTypographies.kt#L20..L24) and then instantiate [it here](https://github.com/xmartlabs/gong/blob/ecbd4871f87338028dd4289eebe55e2d8728b564/app/src/main/java/com/xmartlabs/gong/ui/theme/AppTypographies.kt#L139)


Some time ago I wrote a blog post that explain how to extend the material theme, you can check it here! https://blog.xmartlabs.com/blog/extending-material-theme-in-jetpack-compose/
